front_end/content.py

```diff
@@ -277,10 +296,40 @@ def get_course_id_from_role(self, user_id):
         return row["course_id"]
 
     def add_user(self, user_id, user_dict):
-        sql = '''INSERT INTO users (user_id, user_json)
+        sql = '''INSERT INTO users (user_id, name, given_name, family_name, picture, locale, ace_theme)
+                 VALUES (?, ?, ?, ?, ?, ?, ?)'''
+
+        self.cursor.execute(sql, (user_id, user_dict["name"], user_dict["given_name"], user_dict["family_name"],
+        user_dict["picture"], user_dict["locale"], "tomorrow"))
+
+    def register_user_for_course(self, course_id, user_id):
+        sql = '''INSERT INTO course_registration (course_id, user_id)
                  VALUES (?, ?)'''
 
-        self.cursor.execute(sql, (user_id, json.dumps(user_dict)))
+        self.cursor.execute(sql, (course_id, user_id,))
+
+    def check_user_registered(self, course_id, user_id):
+        sql = '''SELECT *
+                 FROM course_registration
+                 WHERE course_id = ?
+                 AND user_id = ?'''
+
+        self.cursor.execute(sql, (course_id, user_id,))
+        if self.cursor.fetchone():
+            return True
+
+        return False
+
+    def get_user_info(self, user_id):
```

Comment:
This works, but I had something a little different in mind regarding how to store the user information in a context variable. I'll tackle that part after we merge this.